Chim

Chimerik opened the pull request.

Chimerik committed:
Corrections Spells ID

Chimerik committed:
Corrections Cantrip Scaler + ajout commande !walk

Chimerik committed:
Modif diverses

Correction d'un bug affectant les pnjs métamorphosés : ceux-ci perdaient tous leurs buffs dès que leur perso était sauvegardé.

Ajout d'un système qui permet de modifier la position et l'orientation d'un placeable avec les touches du clavier

Chimerik committed:
Custom player

En gros, plein de modifs pour pouvoir avoir une nouvelle classe Player qui hérite de NWPlayer et a ses propres propriétés dotnet

rgranger committed:
Merge branch 'master' into chim

Resoud les conflits

rgranger commented:
J'ai push un commit sur ta branche pour résoudre les conflits avec la branche master

rgranger commented on NWScript/Types/nwobject.cs:
```diff
@@ -3,155 +3,155 @@
+            if (ObjectType != ObjectType.Creature) return;
+            foreach (var currentSlot in Enum.GetValues(typeof(InventorySlot)))
+                NWScript.DestroyObject(NWScript.GetItemInSlot((InventorySlot)currentSlot!, this));
+        }
```
T'as touché à un truc dans ce fichier ? Si non, je pense que c'est juste une histoire de fin de ligne. Il faut que tu configures ton Visual Studio pour mettre des caractères "LF" (linux style) en fin de ligne et pas "CRLF" (windows style).

Ca évite que git détecte une modif sur chaque ligne, alors qu'il n'y a pas eu de modif

rgranger replied:
https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings

Je vais essayer de faire ça pour que ça soit automatique dans ce projet, mais normalement tu peux directement modifier ta config git

rgranger commented on ScriptHandlers/ScriptHandlers.cs:
```diff
@@ -1,448 +1,663 @@
+            return Entrypoints.SCRIPT_HANDLED;
+        }
+    }
+}
```
Idem : on ne peut pas review un truc ou git detecte un changement sur toutes les lignes

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+    {
+        #region Properties
+
+        public readonly int PCID;
```
A quoi correspond cette propriété ? Si c'est l'id, on peut le nommer simplement `id` ou `uid`. Pas besoin de prefixer par pc car on est déjà dans la classe `Player`

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+        #region Properties
+
+        public readonly int PCID;
+        public readonly bool IsDeveloper;
```
En général, par convention, on préfère mettre les properties en `camelCase` et les functions / class en `PascalCase`.

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+        public readonly int PCID;
+        public readonly bool IsDeveloper;
+        public readonly DateTime ConnectionTime = DateTime.Now;
+        public int HeartbeatTicks;
```
Cette propriété à l'air suspecte, à quoi sert-elle?

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+
+        public int Kills;
+        public int KillsAnimal;
+        public int KillsPvP;
```
Hum, du coup à quoi correspond `Kills` ? Si c'est KillsAnimal + KillsPvP, ça ne sert à rien de le stocker dans une variable

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+        {
+            get => _IsSelecting;
+            set => _IsSelecting = value;
+        }
```
Je comprends pas trop ces getter setter ?

Quel est la différence entre :

```cs
public int toto;

// VS
public int toto { get; set; }

// VS
private int _toto;
public int toto { get => _toto; set => _toto = value; }
```

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+           */
+
+            Players[nwobj.AsPlayer().uuid] = this;
+ //           Debug.Info($"Player join: PCID:{PCID}, Name:{Name}, CDKey:{CDKey}, IP:{IPAddress}, BIC:{BicFile}");
```
Un peu chelou comme constructor

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+
+        #region Static
+
+        public static Dictionary<string, Player> Players = new Dictionary<string, Player>();
```
Donc tous les joueurs ont chacun un Dictionnaire avec la liste de tous les joueurs ?

Je pense qu'il y a un truc un peu bizarre, ou alors le nommage est mal choisi et la class `Player` devrait s'appeler `PlayerManager` ou `PlayerSystem` ou qqchose du genre

rgranger commented on Systems/Player/Player.cs:
```diff
@@ -0,0 +1,416 @@
+                   return int.TryParse(nw.Tag, out pcid) ? Players[pcid] : null;
+               }
+          */
+        public static void SendMessageToAllPCs(string msg)
```
On peut définir une meme classe dans plusieurs fichiers avec le mot clef `partial`. Ca permet de ne pas avoir un fichier trop gros, et de split : par exemple la config dans un fichier, les attributs dans un autres, les points d'entrées des scripts dans un autres et les fonctions un peu plus "utilitaires" dans un autre

rgranger commented on Utils/Utils.cs:
```diff
@@ -1,34 +1,43 @@

+        {

+            dmfi_rest=1,
```
Par convention en c#, les enum (et leur valeurs) sont en PascalCase.

Donc ça donnerait ici :

```cs
public enum Meuble
{
  DmfiRest=1,
  MejSocialPlc,
  MejPcSocial,
  Meuble,
  MejPc,
  MejSo
}
```
Btw, c'est mieux de coder en anglais. et à quoi correspondent tous ces nommages ? Je ne comprends pas trop.

Et la valeur `meuble` dans une enum nommée `Meuble` c'est un peu wtf

rgranger replied:
De plus, je ne suis pas sûre que ça soit une bonne idée de mettre des enums dans le fichier Utils.cs. Pour moi ce fichier regroupe plutot des fonctions utilitaires générales (qu'on pourra splitter plus tard en catégories quand il y en aura un certain nombre).

Pour les Enums générales, je rajouterais plutot un fichier Enums/Enums.cs ou qqchose du genre

rgranger committed:
Merge branch 'master' into chim

rgranger committed:
chore(git) : run command git add --renormalize .

rgranger reviewed (commented):
J'ai repush sur ta branche pour normaliser les fins de ligne, mais tu devrais maj ta config git perso en faisant : `git config --global core.autocrlf true`

rgranger committed:
Merge branch 'master' into chim

rgranger committed:
Merge branch 'master' into chim

Chimerik committed:
Auto stash before merge of "chim" and "origin/chim"

rgranger committed:
evolution(mysql) : utilise Dapper pour ameliorer la connexion MySQL

Chimerik committed:
Merge remote-tracking branch 'origin/EV_mysql_dapper' into chim

Chimerik committed:
Modif diverses plus potager

Le potager étant à réécrire en mode objet

Chimerik commented:
bloup

rgranger commented:
Il vaut mieux faire aussi plein de petites MR (Merge Request, ou Pull Request c'est pareil) avec peu de modifs, plutot qu'une grosse MR imbuvable pour le reviewer avec du `+1,148 −350`

En conclusion : split cette grosse branche en petites branches, qui ont chacune une feature spécifique, et ensuite fait une MR par branche dont tu penses la feature "finie", ou en tout cas, utilisable en production. Une grosse feature peut bien sure etre splittée en plusieurs sous-features (et donc en plusieurs branches). Par exemple la feature "Craft" peut etre splittée en features "Forgeron", "Mineur", ... et "Forgeron" peut probablement aussi etre splittée, et ainsi de suite, jusqu'à ce qu'on obtienne des branches de tailles "acceptables" pour un reviewer

Chimerik committed:
Modifs potager nouvelle connexion BDD

Et création de branche

Chimerik committed:
Fin système potager

Abandon du système par uuid qui marche pas comme je pensais.
Retour au système d'id en local int.

Chimerik committed:
Ajout don pour utilisation transformation loup-garou

Chimerik committed:
Merge branch 'potager' into chim
